ipaserver/install/server/replicainstall.py

```diff
@@ -508,22 +508,44 @@ def promote_openldap_conf(hostname, master):
         root_logger.info("Failed to update {}: {}".format(ldap_conf, e))
 
 
-def check_remote_version(api):
+def check_remote_compatibility(api):
+    """
+    Perform a check to verify remote server's version and fips-mode
+
+    :param api: remote api
+
+    :raises: ``ScriptError`` if the checks fails
+    """
     client = rpc.jsonclient(api)
     client.finalize()
 
     client.connect()
     try:
-        env = client.forward(u'env', u'version')['result']
+        env = client.forward(u'env', u'version', u'fips_mode')['result']
     finally:
         client.disconnect()
 
+    # Check version compatibility
     remote_version = parse_version(env['version'])
     api_version = parse_version(api.env.version)
     if remote_version > api_version:
-        raise RuntimeError(
-            "Cannot install replica of a server of higher version ({}) than"
-            "the local version ({})".format(remote_version, api_version))
+        raise ScriptError(
+            "Cannot install replica of a server of higher version "
+            "(%(remote_version)s) than the local version (%(api_version)s)"
+            % dict(remote_version=remote_version, api_version=api_version))
+
+    # Check FIPS mode compatibility
+    remote_fips_mode = env['fips_mode']
+    fips_mode = tasks.is_fips_enabled()
```

Comment:
I think you could rather pass your api.env.fips_mode to `check_remote_compatibility` instead.